presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -1186,6 +1171,18 @@ protected Object visitSubscriptExpression(SubscriptExpression node, Object conte
                 return new SubscriptExpression(toExpression(base, type(node.getBase())), toExpression(index, type(node.getIndex())));
             }
 
+            // Subscript on Row hasn't got a dedicated operator. It is interpreted by hand.
+            if (base instanceof SingleRowBlock) {
+                SingleRowBlock row = (SingleRowBlock) base;
```

Comment:
Are there any tests that hit this path?

Reply:
No, I'll add some to TestExpressionInterpreter.